polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -551,6 +555,66 @@ where
 	Some(processed_code_hashes)
 }
 
+async fn update_active_leaves<Sender>(
+	sender: &mut Sender,
+	mut validation_backend: impl ValidationBackend,
+	update: ActiveLeavesUpdate,
+) where
+	Sender: SubsystemSender<ChainApiMessage> + SubsystemSender<RuntimeApiMessage>,
+{
+	let ancestors = get_block_ancestors(sender, update.activated.as_ref().map(|x| x.hash)).await;
+	if let Err(err) = validation_backend.update_active_leaves(update, ancestors).await {
+		gum::warn!(
+			target: LOG_TARGET,
+			?err,
+			"cannot update active leaves in validation backend",
+		);
+	};
+}
+
+async fn get_allowed_ancestry_len<Sender>(sender: &mut Sender, relay_parent: Hash) -> Option<usize>
+where
+	Sender: SubsystemSender<ChainApiMessage> + SubsystemSender<RuntimeApiMessage>,
+{
+	match prospective_parachains_mode(sender, relay_parent).await {
```

Comment:
We should consider async backing as always enabled, so such code is already obsolete. But let's leave this for the bigger async backing parameters cleanup PRs, so this one makes it to the release.